crossdock/src/tchannel_server.js

```diff
@@ -20,41 +20,61 @@
 // THE SOFTWARE.
 
 import * as constants from './constants.js';
+import ConstSampler from '../../src/samplers/const_sampler.js';
+import Helpers from './helpers';
+import InMemoryReporter from '../../src/reporters/in_memory_reporter.js';
+import Tracer from '../../src/tracer.js';
+import opentracing from 'opentracing';
 import path from 'path';
+import TChannelBridge from '../../src/tchannel_bridge';
 
 import TChannel from 'tchannel';
 import TChannelThrift from 'tchannel/as/thrift';
 import Utils from '../../src/util.js';
 
+let DEFAULT_THRIFT_PATH = '/crossdock/tracetest.thrift';
 export default class TChannelServer {
-    constructor() {
-        let crossdockSpecPath = path.join(__dirname, '..', '..', 'src', 'jaeger-idl', 'thrift', 'crossdock', 'tracetest.thrift');
+    _tracer: Tracer;
+    _helpers: any;
+
+    constructor(crossdockSpecPath=DEFAULT_THRIFT_PATH) {
+        this._tracer = new Tracer('node', new InMemoryReporter(), new ConstSampler(false));
+        this._helpers = new Helpers(this._tracer);
+
         let serverChannel = TChannel({'serviceName': 'node'});
         let tchannelThrift = TChannelThrift({
             'channel': serverChannel,
             'entryPoint': crossdockSpecPath
         });
+        let context = {};
 
-        tchannelThrift.register(serverChannel, 'TracedService::startTrace', {}, this.handleTChannelRequest.bind(this));
-        tchannelThrift.register(serverChannel, 'TracedService::joinTrace', {}, this.handleTChannelRequest.bind(this));
+        let bridge = new TChannelBridge(this._tracer);
+        let tracedHandler = bridge.tracedHandler(this.handleTChannelRequest.bind(this));
+        tchannelThrift.register(serverChannel, 'TracedService::joinTrace', context, tracedHandler);
 
         serverChannel.listen(8082, Utils.myIp(), () => {
             console.log('TChannel server listening on port 8082...');
         });
     }
 
     handleTChannelRequest(context: any, req: any, head: any, body: any, callback: Function) {
-        callback(null, {
+        let isStartRequest: boolean = false;
+        let traceRequest = body.request;
+
+        let promise = this._helpers.handleRequest(
+            isStartRequest,
+            traceRequest,
+            context.openTracingSpan.context(),
```

Comment:
So this is the incredibly strange bug.  If you remove context, and pass in a span then yo go down the code path just accepts a span.  The moment I do that outgoing tchannel requests begin to fail.  What is stranger to me though is that if I modify tchannel_bridge.js to create the context() in the handler then this still fails....